source/library/Autoexporter.hs

```diff
@@ -16,17 +20,35 @@ main = do
 mainWithArgs :: [String] -> IO ()
 mainWithArgs args =
   case args of
-    [name, inputFile, outputFile] -> autoexport name inputFile outputFile
+    [name, inputFile, outputFile, "--deep"] ->
+      autoexport ExportScope'Deep name inputFile outputFile
+    [name, inputFile, outputFile] ->
+      autoexport ExportScope'Shallow name inputFile outputFile
     _ -> fail ("unexpected arguments: " ++ show args)
 
-autoexport :: String -> FilePath -> FilePath -> IO ()
-autoexport name inputFile outputFile = do
+autoexport :: ExportScope -> String -> FilePath -> FilePath -> IO ()
+autoexport exportScope name inputFile outputFile = do
   let moduleName = makeModuleName name
   let directory = FilePath.dropExtension inputFile
-  files <- Directory.getDirectoryContents directory
+  files <- findFiles exportScope directory
   let output = makeOutput moduleName directory files
   writeFile outputFile output
 
+findFiles :: ExportScope -> FilePath -> IO [FilePath]
+findFiles ExportScope'Shallow dir = filter isHaskellFile <$> findRootFiles dir
```

Comment:
Can you refactor this to avoid using `<$>`? I know it's a pretty popular operator, but I try to avoid using it. Either `fmap` or `do` notation is fine.